feat(anta): Add Markdown report option to ANTA (#740)
aristanetworks/anta: docs/cli/nrfu.md

```
![$1anta nrfu json results](../imgs/anta-nrfu-json-output.png){ loading=lazy width="1600" }

## Performing NRFU and saving results in a Markdown file

The `md-report` command in NRFU testing generates a comprehensive Markdown report containing various sections, including detailed statistics for devices and test categories. By default, this command saves all test results from the current run in the Markdown file. However, you can choose to include only failed tests by using the `--only-failed-tests` option in the command line interface.

This feature is particularly useful for comparing reports from multiple runs, allowing you to quickly identify and track issues across your network over time. By focusing on failed tests, you can efficiently pinpoint areas that require attention or have shown improvement between different test executions.

### Command overview

```bash
anta nrfu md-report --help

Usage: anta nrfu md-report [OPTIONS]

  ANTA command to check network state with Markdown report. It only saves test

>   results and not the output from the --group-by option.

Reviewer: Actually I am confused now. `--group-by` is an option for the `table` command so why do we need to specify this?

>   results and not the output from the --group-by option.

Author: Removed in 5c449949773cb86a3c25aa336bd3d83f5150773c

```
![$1anta nrfu json results](../imgs/anta-nrfu-json-output.png){ loading=lazy width="1600" }

## Performing NRFU and saving results in a Markdown file

The `md-report` command in NRFU testing generates a comprehensive Markdown report containing various sections, including detailed statistics for devices and test categories. By default, this command saves all test results from the current run in the Markdown file. However, you can choose to include only failed tests by using the `--only-failed-tests` option in the command line interface.

This feature is particularly useful for comparing reports from multiple runs, allowing you to quickly identify and track issues across your network over time. By focusing on failed tests, you can efficiently pinpoint areas that require attention or have shown improvement between different test executions.

### Command overview

```bash
anta nrfu md-report --help

Usage: anta nrfu md-report [OPTIONS]

  ANTA command to check network state with Markdown report.

Options:
  --md-output FILE     Path to save the report as a Markdown file. It only
                       saves test results and not the output from the --group-

>                        by option.  [env var: ANTA_NRFU_MD_REPORT_MD_OUTPUT;

Reviewer: this should not be there

Author: Fixed in 30c8d7e418e1f273e882bb1dbdc3ac8e7151a4dd

                       required]

>   --only-failed-tests  Flag to determine if only failed tests should be saved
>                        in the report.  [env var:

Reviewer: ```suggestion
  --only-failed-tests  Only include failed tests in the report.
                       [env var:
```

Author: Fixed in 30c8d7e418e1f273e882bb1dbdc3ac8e7151a4dd
